pydata_center/monitoring/permissions.py

```diff
@@ -15,3 +15,20 @@ def has_permission(self, request, view):
             return False
         return request.user.groups.filter(
             name__in=['Admin', 'Operator']).exists()
+
+
+class IsAgentWithPermission(BasePermission):
+    """
+    Custom permission for Agent-based authentication.
+    """
+
+    def has_permission(self, request, view):
+        user = request.user
+
+        # Check if the user is an Agent
+        if hasattr(user, '_meta') and user._meta.model_name == 'agent':
+
+            return user.is_active
+
+        # Otherwise fallback to normal Django permission system (for User)
+        return request.user.has_perm('monitoring.add_serverstatus')
```

Comment:
Consider whether it's appropriate to hardcode a specific permission here. If the permission is view-specific, move this logic into the view itself or configure via `view.required_permission`.
Alternatively, use:
```
getattr(view, 'required_permission', None)
```
This way the permission class is reusable across views.